### PR TITLE
feature: Add new patterns for vulnerability scanning [TAROT-2791]

### PR DESCRIPTION
**Pull request by afsmeira**
You can ignore changes inside the `multiple-tests` folder. These are integration tests that run the tool as an analysis would and given the dependency files will produce the results in the XML.

### CODE REVIEW

**Review by stefanvacareanu7 — APPROVED** [review]

**Comment by afsmeira** [comment]
@lorandszakacs when this is merged and incorporated in `codacy-tools` you will need to change https://github.com/codacy/codacy-worker/pull/868 and add two new patterns: `vulnerability_medium` and `vulnerability_minor`.

**Comment by afsmeira** [comment]
@lventura-codacy please take a look at the documentation and pattern names and suggest any changes you'd like.

File: docs/description/vulnerability.md [hunk 1]
```diff
@@ -1,2 +1,2 @@
-## Insecure dependencies detection
-Detects insecure dependencies by checking the libraries declared in the package manager and flagging used library versions with known security vulnerabilities.
\ No newline at end of file
+## Insecure dependencies detection (critical and high severity)
+Detects critical and high severity insecure dependencies by checking the libraries declared in the package manager and flagging used library versions with known security vulnerabilities.
```
Comment (lventura-codacy) [inline docs/description/vulnerability.md:2]:
```suggestion
Detects insecure dependencies (critical and high severities) by checking the libraries declared in the package manager and flagging used library versions with known security vulnerabilities.
```

File: docs/description/vulnerability_medium.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+## Insecure dependencies detection (medium severity)
+Detects medium severity insecure dependencies by checking the libraries declared in the package manager and flagging used library versions with known security vulnerabilities.
```
Comment (lventura-codacy) [inline docs/description/vulnerability_medium.md:2]:
```suggestion
Detects insecure dependencies (medium severity) by checking the libraries declared in the package manager and flagging used library versions with known security vulnerabilities.
```

File: docs/description/vulnerability_minor.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+## Insecure dependencies detection (minor severity)
+Detects minor severity insecure dependencies by checking the libraries declared in the package manager and flagging used library versions with known security vulnerabilities.
```
Comment (lventura-codacy) [inline docs/description/vulnerability_minor.md:2]:
```suggestion
Detects insecure dependencies (minor severity) by checking the libraries declared in the package manager and flagging used library versions with known security vulnerabilities.
```

**Review by lventura-codacy — COMMENTED** [review]
Some nitpicky suggestions on the docs, otherwise LGTM

**Review by lventura-codacy — APPROVED** [review]

**Pull request merged**